src/domain/core/src/entities/metadata_block_repository.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use async_trait::async_trait;
+use opendatafabric::{MetadataBlock, Multihash};
+
+use crate::{
+    ContainsError,
+    GetBlockError,
+    InsertError,
+    InsertOpts,
+    InsertResult,
+    ObjectRepository,
+};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait]
+pub trait MetadataBlockRepository: Send + Sync {
+    async fn contains(&self, hash: &Multihash) -> Result<bool, ContainsError>;
+
+    async fn get(&self, hash: &Multihash) -> Result<MetadataBlock, GetBlockError>;
+
+    async fn insert<'a>(
```

Comment:
I think each of these methods should include word "block" in the name.

Reply:
Done

Reply:
Now we have:
- `MetadataBlockRepository::get_block()`
- `MetadataBlockRepository::get_block_data()`
- `MetadataBlockRepository::insert_block_data()`